src/core/note_app.dart

```diff
@@ -1,6 +1,109 @@
+// ignore_for_file: unused_local_variable
+
+import 'dart:io';
+import 'utils.dart';
+import '../models/note_model.dart';
+
 class NoteApp {
   // TODO: Complete the run function
   void run() {
-    print('Hello, world!');
+    List<NoteModel> tasks = [];
+    print("♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡");
+    print('Task List App');
+
+    var answer;
+
+    do {
+      print("♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡");
+      print('What would you like to do?');
+      print('1. Add task');
+      print('2. Edit task');
+      print('3. Delete task');
+      print('4. View tasks');
+      print('5. Exit');
+
+      try {
+        answer = int.parse(stdin.readLineSync()!);
+      } catch (e) {
+        print('Invalid input. Please enter a number.');
+      }
+      print("♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡");
+
+      switch (answer) {
+        case 1:
+          String? answer1;
+
+          do {
+            addTask(tasks);
+            print(
+                'Do you want to add another task? Please answer with yes or no:');
+            print("♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡");
+            answer1 = stdin.readLineSync()?.toLowerCase();
+          } while (answer1 == 'yes' || answer1 == 'y');
+
+          break;
+
+        case 2:
+          print('Edit task');
+          print('Enter the task ID you want to edit:');
+          int taskId = int.parse(stdin.readLineSync()!);
+
+          NoteModel? taskToEdit =
+              tasks.firstWhereOrNull((task) => task.id == taskId);
+
+          if (taskToEdit != null) {
+            print('What would you like your new task name to be?');
+            taskToEdit.task_name = stdin.readLineSync()!;
+
+            print('Enter the new task description:');
+            taskToEdit.description = stdin.readLineSync()!;
```

Comment:
add checking if title or description is null
```suggestion
            print('What would you like your new task name to be?');
            String name = stdin.readLineSync()!;

            if(name.isEmpty)
            {
                print("Task name cannot be empty");
                break;
            }
            taskToEdit.task_name = name;



            print('Enter the new task description:');
            taskToEdit.description = stdin.readLineSync()!;
```